symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/test/java/com/symphony/bdk/spring/SymphonyBdkAutoConfigurationTest.java

```diff
@@ -143,4 +143,31 @@ void shouldInstantiateLoadBalancedApiClientIfConfigured() {
       assertThat(config.getAgent().getBasePath()).isEqualTo("http://localhost:443/context");
     });
   }
+
+  @Test
+  void shouldNotInitializeDatafeedIfDisabled() {
+    final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withPropertyValues(
+            "bdk.host=localhost",
+
+            "bdk.bot.username=testbot",
+            "bdk.bot.privateKeyPath=classpath:/privatekey.pem",
+
+            "bdk.datafeed.enabled=false"
+        )
+        .withUserConfiguration(SymphonyBdkMockedConfiguration.class)
+        .withConfiguration(AutoConfigurations.of(SymphonyBdkAutoConfiguration.class));
+
+    contextRunner.run(context -> {
+      assertThat(context).hasSingleBean(SymphonyBdkAutoConfiguration.class);
+
+      final SymphonyBdkCoreProperties config = context.getBean(SymphonyBdkCoreProperties.class);
+      assertThat(config.getAgent().getBasePath()).isEqualTo("https://localhost:443");
+
+      assertThat(context).doesNotHaveBean("datafeedService");
```

Comment:
would it be more robust to test on the class type instead of the name? (in case we rename it/change the bean's name)